packages/launchpad/cypress/e2e/scaffold-project.cy.ts

```diff
@@ -0,0 +1,116 @@
+//
+// To update your expected files, just delete the `expected-cypress` and re-run the test,
+// or modify them by hand.
+function scaffoldAndOpenE2EProject (
+  name: Parameters<typeof cy.scaffoldProject>[0],
+  language: 'js' | 'ts',
+  args?: Parameters<typeof cy.openProject>[1],
+) {
+  cy.scaffoldProject(name)
+  cy.openProject(name, args)
+
+  cy.visitLaunchpad()
+
+  cy.contains('Welcome to Cypress!').should('be.visible')
+  cy.contains('[data-cy-testingtype="e2e"]', 'Not Configured')
+  cy.contains('[data-cy-testingtype="component"]', 'Not Configured')
+  cy.contains('E2E Testing').click()
+  cy.contains(language === 'js' ? 'JavaScript' : 'TypeScript').click()
+  cy.contains('Next').click()
+  cy.contains('We added the following files to your project.')
+  cy.contains('Continue').click()
+  cy.contains('Choose a Browser')
+}
+
+function scaffoldAndOpenCTProject (
+  name: Parameters<typeof cy.scaffoldProject>[0],
+  language: 'js' | 'ts',
+  framework: typeof FRONTEND_FRAMEWORKS[number]['name'],
+  bundler?: typeof FRONTEND_FRAMEWORKS[number]['supportedBundlers'][number]['name'],
+  args?: Parameters<typeof cy.openProject>[1],
+) {
+  cy.scaffoldProject(name)
+  cy.openProject(name, args)
+
+  cy.visitLaunchpad()
+
+  cy.contains('Welcome to Cypress!').should('be.visible')
+  cy.contains('[data-cy-testingtype="e2e"]', 'Not Configured')
+  cy.contains('[data-cy-testingtype="component"]', 'Not Configured')
+  cy.contains('Component Testing').click()
+  cy.contains(language === 'js' ? 'JavaScript' : 'TypeScript').click()
+
+  cy.contains('Pick a framework').click()
+  cy.contains(framework).click()
+  if (bundler) {
+    cy.contains(bundler).click()
+  }
+
+  cy.contains('Next Step').click()
+  cy.contains('Skip').click()
+  cy.contains('We added the following files to your project.')
+  cy.contains('Continue').click()
+  cy.contains('Choose a Browser')
+}
+
+function assertScaffoldedFilesAreCorrect (language: 'js' | 'ts', testingType: Cypress.TestingType) {
+  cy.withCtx((ctx) => ctx.currentProject).then((currentProject) => {
+    cy.task<SnapshotScaffoldTestResult>('snapshotCypressDirectory', {
+      currentProject,
+      testingType,
+      language,
+    })
+    .then((result) => {
+      if (result.status === 'ok') {
+        return result
+      }
+
+      throw new Error(result.message)
+    })
+    .then((res) => {
+      cy.log(`✅ ${res.message}`)
+    })
+  })
+}
+
+describe('scaffolding new projects', () => {
+  it('scaffolds E2E for a JS project', () => {
+    scaffoldAndOpenE2EProject('pristine', 'js')
+    assertScaffoldedFilesAreCorrect('js', 'e2e')
+  })
+
+  it('scaffolds E2E for a TS project', () => {
+    scaffoldAndOpenE2EProject('pristine', 'ts')
+    assertScaffoldedFilesAreCorrect('ts', 'e2e')
+  })
+
+  it('scaffolds CT for a JS project', () => {
+    scaffoldAndOpenCTProject('pristine', 'js', 'Create React App (v5)')
+    assertScaffoldedFilesAreCorrect('js', 'component')
```

Comment:
This will create a folder called `expected-cypress-js-component`, even though it is a `create-react-app` specific scaffold. I don't see this being used more than 4 types per project (2 for testing type, two for language). Should the framework be included in the scaffolded directory or does it make sense to have this per system-test project (where the actual system-test project provides the context of what selection will be chosen)?

Reply:
I see what you are saying. I'm not entirely sure of the best way to handle component setup, since there's lots of different combinations. Tim is also working on some testing infra as part of the CT architecture project - I wonder if we rethink this (the CT specific scaffolding) again once that lands?

My initial idea was we'd use different projects for each CT scaffold test - eg instead of "pristine", we might use "vuecli-unconfigured", so it wouldn't be necessary to have the framework in the name, but now I'm thinking we just use a single project and include the CT framework in the folder too, like you are suggesting. I think I'll make that change now.

Reply:
I've made this change @ZachJW34. https://github.com/cypress-io/cypress/pull/20818/commits/1af1e6c311781ec16a528fc2db2886878454cee8

Reply:
I've seen some of the work Tim is setting up for testing infra. We've got a few systems in place that work well (I like the changes you made to accommodate the CT permutations), curious to see how we can leverage them together with Tim's work.

Reply:
Yep, let's see what improvements we can make overall post CT arch changes landing.